src/box.jl

```diff
@@ -14,6 +14,9 @@ function Box(c::AbstractVector, d::AbstractVector,
     return Box{length(c),typeof(data)}(c, inv(axes ./ sqrt.(sum(abs2,axes,1))), d*0.5, data)
 end
 
+Base.:(==)(b1::Box, b2::Box) = b1.c==b2.c && b1.r==b2.r && b1.p==b2.p && b1.data==b2.data
+Base.hash(b::Box, h::UInt) = hash(b.c, hash(b.r, hash(b.p, hash(b.data, hash(:Box, h)))))
```

Comment:
We could alternatively just add the [AutoHashEquals package](https://github.com/andrewcooke/AutoHashEquals.jl) to the REQUIRES files and do `@auto_hash_equals type Box` etcetera.  But doing it manually should be fine too.